osquery/tables/system/darwin/system_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/database.h>
+#include <osquery/tables.h>
+#include "osquery/core/conversions.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+
+namespace osquery {
+namespace tables {
+
+Status getCpuId(std::string& cpu_id) {
+  auto matching = IOServiceMatching("IOPlatformExpertDevice");
```

Comment:
I considered that, and was going to have `char cpu_id[12+1]` since all the `cpu_id`'s I've seen are all 12 chars, but according to Apple they make no guarantees on the format and length, so I figured I'd just continue dynamically alloc'ing this value.